docs/src/main/asciidoc/security-openid-connect-multitenancy-concept.adoc

```diff
@@ -0,0 +1,317 @@
+////
+This guide is maintained in the main Quarkus repository
+and pull requests should be submitted there:
+https://github.com/quarkusio/quarkus/tree/main/docs/src/main/asciidoc
+////
+= OpenID Connect (OIDC) Multi-Tenancy
+include::_attributes.adoc[]
+:categories: security
+:summary: This overview explains OpenID Connect multi-tenancy for Quarkus applications
+
+The goal of OIDC multi-tenancy is developing an application which can provide functionality to users from different organisations. Each enabled organisation could store its registered users and their credentials in its own `realm`, and use OIDC to supply security-related user information with the application. Each call to the application must therefore contain at least two kinds of information:
+
+* A `tenant identifier` which uniquely identifies the realm where the end user was authenticated / logged in. While you can technically use any means of sending the `tenant identifier` from the user's browser or other client-side application, it is convenient to insert it as a string within the context path of the service.
```

Comment:
This is a very nice, compact summary - but it is only the case, where a tenant id has to be extracted (typically) from the HTTP request URL, is for bearer token authentication requests or `initial` non-authenticated requests to Quarkus `quarkus.oidc.application-type=web-app` endpoints which drive the code flow themselves (example, we'd like users authenticate to Google or Azure, or specific Keycloak realms, etc).

Once a Quarkus `quarkus.oidc.application-type=web-app` user has authenticated for example to Google, then they can access the application URL space without `google` being encoded in URLs - it will be encoded in the session cookie name and Quarkus will itself extract it and make it available to tenant resolves (as a Routing Context `tenant-id` attribute - this is a too technical detail though for this doc)

Reply:
Good feedback, here too. I Will rephrase a tad.

 However - the case of the CORS preflight request comes to mind. The session cookie Will not be sent by a browser to the resource server. So the preflight call Will not be interpreted correctly by the TenantResolver if using a session Cookie.

Am i missing parts of the flow here?

Reply:
@lennartj But the preflight request should end at the CORS filter level and return, so it should work